### PR TITLE
Dearomatize on load #868

### PR DESCRIPTION
**Pull request by even1024**

### CODE REVIEW

File: api/c/indigo/src/indigo_molecule.cpp [hunk 1]
```diff
@@ -508,12 +510,14 @@ CEXPORT int indigoLoadQueryMolecule(int source)
...
         loader.treat_x_as_pseudoatom = self.treat_x_as_pseudoatom;
+        loader.dearomatize_on_load = self.dearomatize_on_load;
+        loader.arom_options = self.arom_options;
 
         std::unique_ptr<IndigoQueryMolecule> molptr = std::make_unique<IndigoQueryMolecule>();
 
         QueryMolecule& qmol = molptr->qmol;
 
-        loader.loadQueryMolecule(qmol);
+        loader.loadMolecule(qmol);
```
Comment (mkviatkovskii) [inline api/c/indigo/src/indigo_molecule.cpp:520]: ?

File: api/c/indigo/src/indigo_reaction.cpp [hunk 2]
```diff
@@ -383,9 +385,11 @@ CEXPORT int indigoLoadQueryReaction(int source)
 
         loader.stereochemistry_options = self.stereochemistry_options;
         loader.treat_x_as_pseudoatom = self.treat_x_as_pseudoatom;
+        loader.dearomatize_on_load = self.dearomatize_on_load;
+        loader.arom_options = self.arom_options;
 
         std::unique_ptr<IndigoQueryReaction> rxnptr = std::make_unique<IndigoQueryReaction>();
-        loader.loadQueryReaction(rxnptr->rxn);
+        loader.loadReaction(rxnptr->rxn);
```
Comment (mkviatkovskii) [inline api/c/indigo/src/indigo_reaction.cpp:392]: ?

File: api/c/indigo/src/indigo_molecule.cpp (hunk 1 above)
Comment (even1024) [reply api/c/indigo/src/indigo_molecule.cpp:520]: loadMolecule now can load both: queries and molecules. Actually we don't need separate methods to do that.

File: api/c/indigo/src/indigo_reaction.cpp (hunk 2 above)
Comment (even1024) [reply api/c/indigo/src/indigo_reaction.cpp:392]: We don't need separate loadQueryReaction. loadReaction can handle both types.

File: core/indigo-core/molecule/molecule_auto_loader.h [hunk 3]
```diff
@@ -38,6 +39,13 @@ namespace indigo
     class QueryMolecule;
     class BaseMolecule;
 
+    enum AutoAromatizeMode
+    {
+        NO_AUTO_AROMATIZE = 0,
+        AUTO_AROMATIZE,
+        AUTO_DEAROMATIZE
+    };
```
[the remark this answers is not in the document]
Comment (mkviatkovskii) [reply core/indigo-core/molecule/molecule_auto_loader.h:47]: Also. if we already define AUTO_AROMATIZE, let's add this option as well. Let's add option "auto-aromatizon-mode" with options "disabled", "aromatize" and "dearomatize", that would be very useful.

[the remark this answers is not in the document]
Comment (even1024) [reply core/indigo-core/molecule/molecule_auto_loader.h]: enum removed

**Pull request merged**